.circleci/config.yml

```diff
@@ -0,0 +1,108 @@
+version: 2.1
+  
+defaults: &defaults
+  working_directory: ~/repo
+  docker:
+    # Jest has issues with node > 16.11.0, please stick node version until the issues are resolved
+    # https://github.com/facebook/jest/issues/11956
+    - image: cimg/node:16.10.0
+
+set_env: &set_env
+  name: Setup Environment Variables
+  command: |
+
+    echo 'Fetching Base Commit from GitHub'
+    echo 'export CIRCLE_PR_NUMBER="${CIRCLE_PR_NUMBER:-${CIRCLE_PULL_REQUEST##*/}}"' >> $BASH_ENV
+    source $BASH_ENV
+    echo "export CIRCLE_PR_BASE_SHA=`curl -s https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/pulls/${CIRCLE_PR_NUMBER} | jq -r '.base.sha'`" >> $BASH_ENV
+    echo 'export AFFECTED_ARGS="--base=${CIRCLE_PR_BASE_SHA}"' >> $BASH_ENV
+    echo 'export NODE_OPTIONS="--max_old_space_size=6144"' >> $BASH_ENV
+
+    source $BASH_ENV
+    echo $AFFECTED_ARGS
```

Comment:
Would you like to get the current PR base to commit SHA and keep `"--base=${CIRCLE_PR_BASE_SHA}"` to  `AFFECTED_ARGS` ( I guess you refer the [nx-example](https://github.com/nrwl/nx-examples/blob/master/.circleci/config.yml) code ) to let the `nx affected` could lint the current PR  changed the part and test the current PR  changed part ( in the next commit )?  

Btw, maybe you could add a comment to hint to us what the part you would like to do for what purpose e.g: get the current PR commit SHA for nx affected could lint and test the only current PR changed code.

Reply:
> Would you like to get the current PR base to commit SHA and keep "--base=${CIRCLE_PR_BASE_SHA}" to AFFECTED_ARGS ( I guess you refer the [nx-example](https://github.com/nrwl/nx-examples/blob/master/.circleci/config.yml) code ) to let the nx affected could lint the current PR changed the part and test the current PR changed part ( in the next commit )?

I don't quite get what you mean 🤣 , did you mean what's the reason of saving base SHA? Yes, I want to let CI only test/lint the changed packages, i.e. the difference from HEAD to base.

> Btw, maybe you could add a comment to hint to us what the part you would like to do for what purpose e.g: get the current PR commit SHA for nx affected could lint and test the only current PR changed code.

Added.

Reply:
Yes, it my question 😆, thanks for still explaining to me 👍